player/src/main.rs

```diff
@@ -0,0 +1,559 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*! This is a player for WebGPU traces.
+ *
+ * # Notes
+ * - we call device_maintain_ids() before creating any refcounted resource,
+ *   which is basically everything except for BGL and shader modules,
+ *   so that we don't accidentally try to use the same ID.
+!*/
+
+use wgc::device::trace;
+
+use std::{
+    ffi::CString,
+    fmt::Debug,
+    fs::File,
+    marker::PhantomData,
+    path::{Path, PathBuf},
+    ptr,
+};
+
+macro_rules! gfx_select {
```

Comment:
Should we expose `gfx_select` from wgpu-core and use that?

Reply:
possibly, yes. Today, the one used in wgpu-rs is slightly different though. We may end up with small differences across the users, and the macro itself is pretty small. Let's keep an eye on it for follow-ups?